sirepo/reply.py

```diff
@@ -367,13 +377,105 @@ def render_static_jinja(self, base, ext, j2_ctx, cache_ok=False):
+                    "Cache-Control",
+                    "private, max-age={resp.cache_control.max_age}",
+                )
+                if c.mtime is not None:
+                    resp.set_header(
+                        "Last-Modified",
+                        email.utils.formatdate(c.mtime, usegmt=True),
+                    )
+            else:
+                resp.set_header("Cache-Control", "no-cache, no-store, must-revalidate")
+                resp.set_header("Pragma", "no-cache")
+
+        def _content_type(resp):
+            c = a.content_type
+            if c in _MIME_TYPE_UTF8:
+                c += '; charset="utf8"'
+            r.set_header("Content-Type", c)
+
+        def _cookie(resp):
+            # TODO(robnagler) http.cookies 3.8 introduced samesite and blows up otherwise.
+            # we know how our cookies are formed so
+            a = self.__attrs
+            if not ("cookie" in a and 200 <= a.status < 400):
+                return
+            c = a.cookie
+            r = [f'{c.pkdel("key")}={c.pkdel("value")}', "Path=/"]
+            for k in sorted(c.keys()):
+                if k == "httponly":
+                    if c[k]:
+                        r.append("HttpOnly")
+                elif k == "max_age":
+                    r.append(f"Max-Age={c[k]}")
+                elif k == "samesite":
+                    r.append(f"SameSite={c[k]}")
+                elif k == "secure":
+                    if c[k]:
+                        r.append("Secure")
+                else:
+                    raise AssertionError(f"unhandled cookie attr={k}")
+            resp.set_header("Set-Cookie", "; ".join(r))
+
+        def _file(resp):
+            a = self.__attrs
+            c = a.content
+            resp.write(c.handle.read())
+            if _DISPOSITION not in a.headers:
+                self._disposition("inline")
+            resp.set_header(
+                "Last-Modified",
+                email.utils.formatdate(c.mtime, usegmt=True),
+            )
+            if c.encoding:
+                resp.set_header("Content-Encoding", c.encoding)
+            resp.set_header("Content-Length", str(c.length))
+
+        a = self.__attrs
+        c = a.get("content")
+        if c is None:
+            c = b""
+            a.content_type = _MIME_TYPE.txt
+        r = self.internal_req
+        a.pksetdefault(status=200)
+        if isinstance(c, PKDict):
+            _file(r)
+        else:
+            _bytes(r, c)
+        r.set_status(a.status)
+        for k, v in a.headers.items():
+            r.set_header(k, v)
+        _content_type(r)
+        _cache_control(r)
+        _cookie(r)
+        return r
+
+    def _disposition(self, disposition):
+        self.header_set(
+            _DISPOSITION, f'{disposition}; filename="{self.__attrs.download_name}"'
         )
         return self
 
+    def _download_name(self, filename):
+        f = re.sub(r"[^\w\.]+", "-", filename).strip("-")
```

Comment:
A name for the rexeg would be helpful to understand why it is being performed.

Reply:
d57937d0